src/geometric_utils.ts

```diff
@@ -126,17 +214,178 @@ export class GeometricUtils {
+                // Ensure both polygons are closed
+                if (GeometricUtils.is_polygon_closed(poly1) && GeometricUtils.is_polygon_closed(poly2)) {
+                    let intersection: ULabelSpatialPayload2D = GeometricUtils.get_polygon_intersection_single(poly1, poly2);
+                    if (intersection != null) {       
+                        // Don't add duplicate intersections
+                        let is_duplicate: boolean = false;
+                        for (let k: number = 0; k < ret.length; k++) {
+                           if (GeometricUtils.polygons_are_equal(ret[k], intersection)) {
+                               is_duplicate = true;
+                               break;
+                           }
+                        }
+                        if (!is_duplicate) {
+                            ret.push(intersection);
+                        }
+                    }
+                }
+            }
+        }
+        return ret;
+    }
+
+    // Return the intersection of two polygons
+    public static get_polygon_intersection_single(poly1: ULabelSpatialPayload2D, poly2: ULabelSpatialPayload2D): ULabelSpatialPayload2D {
+        // Convert to turf polygons
+        try {
+            let poly1_turf = turf.polygon([poly1]);
+            let poly2_turf = turf.polygon([poly2]);
+            
+            // Find intersection
+            let intersection = turf.intersect(poly1_turf, poly2_turf);
+            if (intersection === null) {
+                return null;
+            } else {
+                // Convert back to ULabelSpatialPayload2D
+                return intersection.geometry.coordinates[0];
+            }
+        } catch (e) {
+            return null;
+        }
+    }
+
+    // Check if polygon is closed, i.e. first and last points are the same and there are at least 3 points
+    public static is_polygon_closed(poly: ULabelSpatialPayload2D): boolean {
+        let ret: boolean = false;
+        if (poly.length > 2) {
+            try {
+                ret = poly[0][0] === poly.at(-1)[0] && poly[0][1] === poly.at(-1)[1];
+            } catch (e) {}
+        }
+        return ret;
+    }
+
+    // Check if two polygons are equal
+    public static polygons_are_equal(poly1: ULabelSpatialPayload2D, poly2: ULabelSpatialPayload2D): boolean {
+        // Check if the polygons have the same number of points
+        if (poly1.length !== poly2.length) {
+            return false;
+        }
+        // Check that each point in poly1 is in poly2
+        for (let i: number = 0; i < poly1.length; i++) {
+            let found: boolean = false;
+            for (let j: number = 0; j < poly2.length; j++) {
+                if (poly1[i][0] === poly2[j][0] && poly1[i][1] === poly2[j][1]) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Similar to the above `line_segments_are_equal` suggestion:
```suggestion
    return _.isEqual(poly1, poly2)
```

Reply:
Removed in [ef7fda8](https://github.com/SenteraLLC/ulabel/commit/ef7fda8847ae8f262b2915dd0127f3066b0111dd)